### PR TITLE
Remove default values for proxy and chainID

### PR DESCRIPTION
**Pull request by popenta**
Removed the default values for the `chainID` and for the `proxy`. Now if `--proxy` is provided will get the `chainID` from the network. If both are provided and the values do not match, will use the value obtained from the proxy.

### CODE REVIEW

File: multiversx_sdk_cli/cli_shared.py [hunk 1]
```diff
@@ -75,8 +78,8 @@ def add_tx_args(args: List[str], sub: Any, with_nonce: bool = True, with_receive
     if with_data:
         sub.add_argument("--data", default="", help="the payload, or 'memo' of the transaction (default: %(default)s)")
 
-    sub.add_argument("--chain", default=config.get_chain_id(), help="the chain identifier (default: %(default)s)")
-    sub.add_argument("--version", type=int, default=config.get_tx_version(), help="the transaction version (default: %(default)s)")
+    sub.add_argument("--chain", help="the chain identifier")
+    sub.add_argument("--version", type=int, default=2, help="the transaction version (default: %(default)s)")
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_shared.py:82]: We can define a constant here.

File: multiversx_sdk_cli/cli_shared.py [hunk 2]
```diff
@@ -172,6 +175,28 @@ def prepare_nonce_in_args(args: Any):
         args.nonce = account.nonce
 
 
+def prepare_chain_id_in_args(args: Any):
+    if not args.chain and not args.proxy:
+        raise ArgumentsNotProvidedError("chain or proxy should be provided")
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_shared.py:180]: `chain ID cannot be inferred / decided: --chain or --proxy should be provided` (alternative).

File: multiversx_sdk_cli/cli_shared.py [hunk 3]
```diff
@@ -172,6 +175,28 @@ def prepare_nonce_in_args(args: Any):
...
+
+    if args.chain and args.proxy:
+        proxy = ProxyNetworkProvider(args.proxy)
+        chain_id = proxy.get_network_config().chain_id
+
+        if args.chain != chain_id:
+            show_deprecation_warning("The chainID you have provided does not match the chainID you got from the proxy. Will use the value from the proxy.")
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_shared.py:187]: `chain ID`.

Here, it should somehow be `show_warning` (it is not a deprecation warning per se).

File: multiversx_sdk_cli/cli_shared.py [hunk 4]
```diff
@@ -172,6 +175,28 @@ def prepare_nonce_in_args(args: Any):
         args.nonce = account.nonce
 
 
+def prepare_chain_id_in_args(args: Any):
+    if not args.chain and not args.proxy:
+        raise ArgumentsNotProvidedError("chain or proxy should be provided")
+
+    if args.chain and args.proxy:
+        proxy = ProxyNetworkProvider(args.proxy)
+        chain_id = proxy.get_network_config().chain_id
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_shared.py:184]: Perhaps rename to `fetched_chain_id` (or something similar).

File: multiversx_sdk_cli/cli_shared.py [hunk 5]
```diff
@@ -172,6 +175,28 @@ def prepare_nonce_in_args(args: Any):
...
+
+        if args.chain != chain_id:
+            show_deprecation_warning("The chainID you have provided does not match the chainID you got from the proxy. Will use the value from the proxy.")
+            args.chain = chain_id
+            return
+        else:
+            return
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_shared.py:191]: Add a comment, perhaps? `if the CLI-provided chain ID is correct, we do not patch the arguments`.

File: multiversx_sdk_cli/cli_transactions.py [hunk 6]
```diff
@@ -65,6 +67,7 @@ def create_transaction(args: Any):
     cli_shared.check_guardian_and_options_args(args)
     cli_shared.check_broadcast_args(args)
     cli_shared.prepare_nonce_in_args(args)
+    cli_shared.prepare_chain_id_in_args(args)
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_transactions.py:70]: I think we have to check all occurrences of `prepare_nonce_in_args` etc. and accompany them with the new `prepare_chain_id_in_args`?

File: multiversx_sdk_cli/tests/test_shared.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+from multiversx_sdk_cli.cli_shared import prepare_chain_id_in_args
+from multiversx_sdk_cli.errors import ArgumentsNotProvidedError
+
+
+class Args:
+    pass
+
+
+def test_prepare_chain_id():
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/tests/test_shared.py:13]: :+1:

Comment (andreibancioiu) [inline multiversx_sdk_cli/tests/test_shared.py:13]: `test_prepare_chain_id_in_args()`

File: multiversx_sdk_cli/cli_shared.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py:82]: created a constant `DEFAULT_TX_VERSION = 2`

File: multiversx_sdk_cli/cli_shared.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py:180]: Replaced.

File: multiversx_sdk_cli/cli_shared.py (hunk 3 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py:187]: renamed the method to `show_warning`

File: multiversx_sdk_cli/cli_shared.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py:184]: renamed to `fetched_chain_id`

File: multiversx_sdk_cli/cli_shared.py (hunk 5 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py:191]: Added

File: multiversx_sdk_cli/tests/test_shared.py (hunk 7 above)
Comment (popenta) [reply multiversx_sdk_cli/tests/test_shared.py:13]: Renamed

**Review by andreibancioiu — DISMISSED** [review]

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk_cli/cli_shared.py [hunk 8]
```diff
@@ -172,6 +176,28 @@ def prepare_nonce_in_args(args: Any):
...
+def prepare_chain_id_in_args(args: Any):
+    if not args.chain and not args.proxy:
+        raise ArgumentsNotProvidedError("chain ID cannot be decided: `--chain` or `--proxy` should be provided")
+
+    if args.chain and args.proxy:
+        proxy = ProxyNetworkProvider(args.proxy)
+        fetched_chain_id = proxy.get_network_config().chain_id
+
+        if args.chain != fetched_chain_id:
+            show_warning("The chain ID you have provided does not match the chain ID you got from the proxy. Will use the value from the proxy.")
```
Comment (bogdan-rosianu) [inline multiversx_sdk_cli/cli_shared.py:188]: does not match the chain ID retrieved from the proxy. Will use the proxy's value (%s)

where %s is the actual chain id

File: multiversx_sdk_cli/constants.py [hunk 9]
```diff
@@ -6,3 +6,6 @@
 DEFAULT_CARGO_TARGET_DIR_NAME = "default_cargo_target"
 TRANSACTION_OPTIONS_TX_GUARDED = 0b0010
 ONE_YEAR_IN_SECONDS = 60 * 60 * 24 * 365
+
+
+DEFAULT_TX_VERSION = 2
```
Comment (bogdan-rosianu) [inline multiversx_sdk_cli/constants.py:11]: are 2 empty lines needed?

Comment (popenta) [reply multiversx_sdk_cli/constants.py:11]: no, left just one empty line

File: multiversx_sdk_cli/cli_shared.py (hunk 8 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_shared.py]: Done!

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**